### PR TITLE
Add layout management

### LINKED ISSUE #4: Using the Layout Engine
Hi !
It would be nice to change the code to use the layout engine instead of hardcoding to improve the UI.
Thanks!

### LINKED ISSUE #5: Add localization
Randomizer can have localization to use the layout engine (Issue https://github.com/HaikuArchives/Randomizer/issues/4). This kind of depends on the former for proper display as some languages tends to overflow fixed-size controls.

### PR DESCRIPTION
**Pull request by janus2**
* Add localization
* Fixes #5
* Fixes #4

### CODE REVIEW

**Comment by humdingerb** [comment]
I'm not super fond of hiding the generated password widgets before hitting the Generate button.
In fact, it'd be nice if a password would be generated automatically when the app launches. Then it'd be: Launch Randomizer -> ALT+C -> ALT+Q and you're done.
(After a menu "Copy to clipboard (ALT+C)" is added, of course. :)

I do like the copy button. Maybe add a tool tip "Copy to clipboard" there, just in case.
I find the tool tip helpful for Symbols and Custom, not so much for the rest of the options which are pretty obvious.

Nice work in any case!

**Comment by janus2** [comment]
> Launch  Randomizer -> ALT+C -> ALT+Q and you're done.

I think to be useful the application should save the user settings.

**Comment by humdingerb** [comment]
> I think to be useful the application should save the user settings.

Definitely!

File: src/AppWindow.cpp [hunk 1]
```diff
@@ -151,19 +178,154 @@ void AppWindow::SetupMenuBar()
...
+}
+
+void
+AppWindow::UnarchivePreferences()
+{	BMessage message;
+	LoadPreferences(message);
+
+	int32 length;
+	if (message.FindInt32("PassLength", &length) == B_OK)
+	{
```
Comment (humdingerb) [inline src/AppWindow.cpp]: If we aim for Haiku code style, no curlies {} needed. Also below ofc.

File: src/AppWindow.cpp [hunk 2]
```diff
@@ -151,19 +178,154 @@ void AppWindow::SetupMenuBar()
...
+		NumCB->SetValue(controlOn ? B_CONTROL_ON : B_CONTROL_OFF);
+	}
+	if (message.FindBool("SpecSymbCB", &controlOn) == B_OK)
+	{
+		SpecSymbCB->SetValue(controlOn ? B_CONTROL_ON : B_CONTROL_OFF);
+	}
+	if (message.FindBool("CustSymbCB", &controlOn) == B_OK)
+	{
+		CustSymbCB->SetValue(controlOn ? B_CONTROL_ON : B_CONTROL_OFF);
```
Comment (humdingerb) [inline src/AppWindow.cpp:306]: Opening { on previous line.

**Review by humdingerb — COMMENTED** [review]
Just some code styling... :)

File: src/AppWindow.cpp (hunk 1 above)
Comment (janus2) [reply src/AppWindow.cpp]: If we aim for Haiku code style there are variables name conventions, functions name conventions... where we stop.

Comment (humdingerb) [reply src/AppWindow.cpp]: Whereever you draw the line. :)

**Pull request merged**